scalding-core/src/main/scala/com/twitter/scalding/typed/Sketched.scala

```diff
@@ -121,6 +129,17 @@ case class SketchJoined[K: Ordering, V, V2, R](left: Sketched[K, V],
       .withReducers(numReducers)
       .map{ case ((r, k), v) => (k, v) }
   }
+
+  private implicit def intKeyOrd: Ordering[(Int, K)] = {
+    val kord = implicitly[Ordering[K]]
+
+    kord match {
+      case kos: OrderedSerialization[_] => new OrderedSerialization2(ordSer[Int], kos.asInstanceOf[OrderedSerialization[K]])
+      case _ => Ordering.Tuple2[Int, K]
+    }
+  }
+
+  private implicit def ordSer[T]: OrderedSerialization[T] = macro OrderedSerializationProviderImpl[T]
```

Comment:
right... you can put this method in scalding-serialization-macros and call it from scalding-core fine though